tests/unit/restapi/v1/test_queue.py

```diff
@@ -0,0 +1,516 @@
+
+
+@pytest.mark.v1
+def test_queue_get_all(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_queues: dict[str, Any],
+) -> None:
+    """Test that all queues can be retrieved.
+
+    Given an authenticated user and registered queues, this test validates the following
+        sequence of actions:
+
+    - A user registers three queues, "tensorflow_cpu", "tensorflow_gpu", "pytorch_cpu".
+    - The user is able to retrieve a list of all registered queues.
+    - The returned list of queues matches the full list of registered queues.
+    """
+    queue1_expected = registered_queues["queue1"].get_json()
+    queue2_expected = registered_queues["queue2"].get_json()
+    queue3_expected = registered_queues["queue3"].get_json()
+    queue_expected_list = [queue1_expected, queue2_expected, queue3_expected]
+
+    assert_retrieving_all_queues_works(client, expected=queue_expected_list)
+
+
+@pytest.mark.v1
+def test_queue_search_query(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_queues: dict[str, Any],
+) -> None:
+    """Test that queues can be queried with a search term.
+
+    Given an authenticated user and registered queues, this test validates the following
+        sequence of actions:
+
+    - The user is able to retrieve a list of all registered queues with a description
+      that contains 'queue'.
+    - The returned list of queues matches the expected matches from the query.
+    """
+    queue1_expected = registered_queues["queue1"].get_json()
+    queue2_expected = registered_queues["queue2"].get_json()
+    queue_expected_list = [queue1_expected, queue2_expected]
+
+    # TODO the query field must be updated when the grammar for searches is updated.
+    search_parameters = {"query": "description:*queue*"}
+
+    assert_retrieving_all_queues_works(
+        client, expected=queue_expected_list, Q=search_parameters
+    )
+
+
+@pytest.mark.v1
+def test_queue_group_query(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_queues: dict[str, Any],
+) -> None:
+    """Test that queues can retrieved using a group filter.
+
+    Given an authenticated user and registered queues, this test validates the following
+        sequence of actions:
+
+    - The user is able to retrieve a list of all registered queues that are owned by the
+      default group.
+    - The returned list of queues matches the expected list owned by the default group.
+    """
+    queue1_expected = registered_queues["queue1"].get_json()
+    queue2_expected = registered_queues["queue2"].get_json()
+    queue3_expected = registered_queues["queue3"].get_json()
+    queue_expected_list = [queue1_expected, queue2_expected, queue3_expected]
+
+    search_parameters = {"group_id": auth_account["default_group_id"]}
+
+    assert_retrieving_all_queues_works(
+        client, expected=queue_expected_list, G=search_parameters
+    )
```

Comment:
The `G` argument is shorthand for a `group_id` query, right? If so, we shouldn't need to provide the `group_id` key in a dict, the assert function can build it for us.